cmd/otelcol/Dockerfile

```diff
@@ -1,16 +1,34 @@
-FROM alpine:3.12 as certs
+ARG ALPINE_VERSION=3.12
+
+FROM alpine:${ALPINE_VERSION} as certs
 RUN apk --update add ca-certificates
 
-FROM alpine:3.12 AS otelcol
+FROM alpine:${ALPINE_VERSION} AS otelcol
 COPY otelcol /
 # Note that this shouldn't be necessary, but in some cases the file seems to be
 # copied with the execute bit lost (see #1317)
 RUN chmod 755 /otelcol
 
+FROM alpine:${ALPINE_VERSION} AS smartagent
+ARG SMART_AGENT_RELEASE_URL
+RUN  wget -O /signalfx-agent.tar.gz "$SMART_AGENT_RELEASE_URL" && \
+     tar -xzf /signalfx-agent.tar.gz && \
+     /signalfx-agent/bin/patch-interpreter /signalfx-agent && \
+     rm -f /signalfx-agent/bin/signalfx-agent /signalfx-agent/bin/agent-status /signalfx-agent.tar.gz
+
 FROM scratch
 COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
 COPY --from=otelcol /otelcol /
-COPY config/collector/gateway_config.yaml /etc/otel/collector/gateway_config.yaml
+
+ENV SPLUNK_BUNDLE_DIR="/usr/lib/splunk-otel-collector/agent-bundle"
+COPY --from=smartagent /signalfx-agent $SPLUNK_BUNDLE_DIR
+ENV SPLUNK_COLLECTD_DIR=$SPLUNK_BUNDLE_DIR/run/collectd
+
+COPY config/collector/gateway_config.yaml    /etc/otel/collector/gateway_config.yaml
 COPY config/collector/otlp_config_linux.yaml /etc/otel/collector/otlp_config_linux.yaml
+COPY config/collector/agent_config.yaml      /etc/otel/collector/agent_config.yaml
+
+ENV SPLUNK_CONFIG=/etc/otel/collector/agent_config.yaml
```

Comment:
@jchengsfx @jrcamp can you confirm this is desired?

Reply:
If so I think we need to set a number of default env vars here.  May be best to defer to a subsequent PR.

Reply:
Yes, the agent config requires more env vars to be set.  Not sure if we want to change the default config from gateway to agent, but maybe @jrcamp will know best.

Reply:
I'll to remove this env var.